src/passes/SeparateDataSegments.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Emits the data segments to a file. The file contains data from address base
+ * onwards (we must pass in base, as we can't tell it from the wasm - the
+ * first segment may start after a run of zeros, but we need those zeros in
+ * the file).
```

Comment:
Please move this to a `//` comment below. We don't merge the description into the license - we keep the license in a uniform way.